process/process.go

```diff
@@ -18,7 +18,6 @@ import (
 
 var (
 	invoke                 common.Invoker = common.Invoke{}
-	ErrorNoChildren                       = errors.New("process does not have children")
```

Comment:
I understand that `ErrorNoChildren` is a practically impossible error definition because when there are no children, `pgrep` exits with status 1. Therefore, it would be fine to remove this definition. However, since this definition has already been published, removing it would cause compilation failures.

So, how about adding a comment like `Deprecated: This error is practically non-occurring`? This way, it will be noted in the [documentation](https://go.dev/wiki/Deprecated) and [flagged by some linters](https://staticcheck.dev/docs/checks/#SA1019).

Reply:
Do you suggest to still keep the variable definition, but to never return it when there are no child processes found? Or to still return it when there are no child processes?

Keep in mind the API contract was already broken as this error was never documented in the Children() function comment and more importantly it was only returned on linux.

Reply:
> Do you suggest to still keep the variable definition, but to never return it when there are no child processes found? Or to still return it when there are no child processes?

I am thinking of keeping this variable defined, and an empty list and `nil` would be returned. It means `ErrorNoChildren` is never used.


> Keep in mind the API contract was already broken as this error was never documented in the Children() function comment and more importantly it was only returned on Linux.

You are indeed correct that the error is not mentioned in the `Children()` function comment. However, unfortunately, [pkg.go.dev](https://pkg.go.dev/github.com/shirou/gopsutil/v4@v4.24.8/process#pkg-variables) displays the Linux definition by default, and this variable is included in that documentation. There may be users who are only looking at this variable and use it.

Reply:
Sure, done.